Lib/test/test_capi/test_opt.py

```diff
@@ -2637,6 +2637,17 @@ def f2():
 
         f2()
 
+    def test_next_instr_for_exception_handler_set(self):
+        # gh-140104: We just want the exception to be caught properly.
+        def f():
+            for i in range(TIER2_THRESHOLD + 3):
+                try:
+                    g(i)
```

Comment:
```suggestion
                    undefined_variable(i)
```
I'm assuming that the exception here is that `g` is undefined?


Can we have two tests?
1. where the exception is raised on the first instruction of the block (like this one), and
2. where it is raised on the last instruction.
Something like:
```
      a_list[""] = 0
```

Reply:
I don't quite understand how to raise something on the last instruction. What does `a_list` contain?

Reply:
`a_list` contains a list.

`a_list[""] = 0`
compiles to 
```
              LOAD_SMALL_INT           0
              LOAD_FAST_BORROW         0 (a_list)
              LOAD_CONST               1 ('')
              STORE_SUBSCR
```
The `STORE_SUBSCR` will raise an exception.

Reply:
I have to admit I straight up forgot how to code Python and thought that was valid Python code that wouldn't raise an exception.